### PR TITLE
Added the ability to select a database sqlite driver in the config

### PR DESCRIPTION
**Pull request by Insei**
Signed-off-by: Aleksandrov Dmitriy <goodmobiledevices@gmail.com>

### CODE REVIEW

File: src/infrastructure/GormDbInitializer.go [hunk 1]
```diff
@@ -22,6 +26,55 @@ func (ns OurNamingSchema) TableName(str string) string {
...
+}
+
+func newGormDb(cfg domain.DbConfig) (*gorm.DB, error) {
+	switch cfg.Driver {
+	case "mysql":
+		return createMySQLDb(cfg.MySQL)
+	case "sqlite":
+		return createSQLiteDb(cfg.SQLite)
+	default:
+		return nil, errors.Internal.New("unsupported db driver")
```
Comment (mikem-zed) [inline src/infrastructure/GormDbInitializer.go]: the error is misleading. Should print "no db driver specified" if the DB config is missing

**Review by mikem-zed — CHANGES_REQUESTED** [review]
except on comment LGTM

File: src/appConfig.yml [hunk 2]
```diff
@@ -4,24 +4,35 @@ httpServer:
...
 database:
   entity:
-    dbName: "roldb"
-    username: "root"
-    password: "67Edh68Tyt69"    # pragma: allowlist secret
-    protocol: "tcp"
-    hostname: "localhost"
-    port: "3306"
-    parameters: "?charset=utf8mb4&parseTime=True&loc=Local"
+    driver: "mysql"
```
Comment (mikem-zed) [inline src/appConfig.yml]: let's set SQLite the default one for now

**Review by mikem-zed — CHANGES_REQUESTED** [review]

File: src/infrastructure/GormDbInitializer.go (hunk 1 above)
Comment (Insei) [reply src/infrastructure/GormDbInitializer.go]: I think that in general everything is written correctly, but I can change this section to a more extended one.
```
case "": 
       return nil, errors.Internal.New("no db driver specified")
default:
       return nil, errors.Internal.Newf("unsupported db driver: %s" cfg.Driver)
```
@mikem-zed Will it fit?

File: src/appConfig.yml (hunk 2 above)
Comment (Insei) [reply src/appConfig.yml]: Okay

**Review by mikem-zed — APPROVED** [review]
LGTM

**Pull request merged**